terratest/test/helpers.go

```diff
@@ -47,8 +57,8 @@ func GetIngressIPs(t *testing.T, options *k8s.KubectlOptions, ingressName string
 }
 
 // Dig gets sorted slice of records related to dnsName
-func Dig(t *testing.T, dnsServer string, dnsPort int, dnsName string) ([]string, error) {
-	port := fmt.Sprintf("-p%v", dnsPort)
+func Dig(t *testing.T, dnsServer string, dnsPort string, dnsName string) ([]string, error) {
```

Comment:
I know it is out of the scope of this task, just reacting to @somaritane note about `mikeg dns`, maybe we can reuse gopkg, so implementation would look like this:
```go
func Dig(t *testing.T, dnsServer string, dnsPort string, dnsName string) ([]string, error) {
	return dns.Dig(fmt.Sprintf("%s:%v", dnsServer,dnsPort),dnsName)
}
```

or we can completelly remove `Dig` function in helper

Reply:
@somaritane @kuritka what is the motivation to use miekg dns in the e2e tests? Not strongly against it, just trying to understand. For me `dig` invocation is directly  reproducible in the CLI and I see a value in it.

Reply:
@ytsarev , from my side: Main reason, it is reusing gopkg which reduces complexity of `Dig` function in `helpers.go`. 

gopkg abstracts implementation details, but internally it uses mikeg.

Reply:
@ytsarev, for me motivation is to be able to use programmatic API, instead of external shell command and parse its results, which might be cumbersome when we test some complex cases. 
Also in this case we don't have to ensure that test run environment has `dig` installed.

Reply:
@somaritane I might be crushing fancy golang party here but the whole `terratest` as a tool is a wrapper around shell commands :D And it is for a reason. We are writing e2e tests mimicking customer behavior, not extending the core controller.

Reply:
@ytsarev I tend to disagree here. 
Speaking about `terratest`: while it offers wrappers for commands like `helm` and `kubectl`, its functions for k8s primitives are using [k8s apimachinery](https://github.com/gruntwork-io/terratest/blob/master/modules/k8s/ingress.go#L19), it's cloud providers are using respective go apis and sdks. 
I guess, this is also for a reason. It would be not feasible to parse CLI response every time, even in json format.

As I previously pointed out [before](https://github.com/AbsaOSS/k8gb/pull/452#discussion_r616756472), similar helper in https://github.com/AbsaOSS/coredns-crd-plugin/blob/main/terratest/test/helper.go#L35 has been rewritten in order to use `github.com/miekg/dns` instead of `dig` wrapper, and we were discussing about consolidating the code in `gopkg`. Do you consider this bad idea? Should it be rewritten back?
`miek/dns` allowed to efficiently test for proper RCODE in response for `AAAA` query, as well as resolved IPs, instead of parsing `dig` output for both.

https://github.com/miekg/dns is pretty stable and mature project, supporting many DNS-related RFCs, having wide community of [users](https://github.com/miekg/dns#users), so I don't know how it maybe inferior to pure `dig`.

And if we talk about real e2e tests mimicking customer behavior, then we should be using something close to the browser or http client and check if we really hit test app endpoint with particular IP address.

Reply:
@somaritane No need to rewrite anything and I agree almost on all points.

There was no real need to sell me https://github.com/miekg/dns as it is the main library of coredns which is an integral part of k8gb from day 0. 

Also, we already use it for k8gb testing in places where it makes direct sense, e.g. https://github.com/AbsaOSS/k8gb/blob/master/controllers/fakedns.go#L27

`coredns-crd-plugin` also had a proper use case for it.

But in this specific test suite, I do not observe the need except for pure aesthetics reasons at the moment.

When I execute the test suite I see output like
```sh
 Running command dig with args [-p5053 @localhost terratest-failover.k8gb-test-nonprod.cloud.example.com +short]
```
If something goes wrong I can directly copy-paste it to my shell and run `dig` without any additional cognitive load of params substitution.

And I'm pretty happy with this flow.

Reply:
@ytsarev In your initial comment https://github.com/AbsaOSS/k8gb/pull/452#discussion_r617635483, you've asked for reasoning about using programmatic api vs command-line wrapper. @kuritka and I provided them.
It feels like at the end you have strong argument to use dig wrapper.
But imho it is not close to mimicking e2e customer behavior, customer use case is not to run `dig` CLI with @localhost as resolver, but rather access service endpoint via browser or http client.

Reply:
@somaritane sure, but there are different k8gb customers I guess. Here we are focusing more on the ones who operate k8gb, not consuming the app. Dig-based tests are checking DNS response correctness the same way as the person validating the lb strategy in action just in an automated way... Curling would not expose the same level of detailed check.